README.md

```diff
@@ -1,133 +1,616 @@
+  }));
+  ```
+- Models: Types and interfaces exported from `models/index.ts`, found individually in `models/model_$N.ts`, and errors including a base `$SERVICEServicexception.ts`.
+  ```typescript
+  import {
+    $SERVICEClient,
+    $SERVICEServiceException,
+    $OPERATIONCommand,
+    $OPERATIONCommandOutput
+  } from "..."; // example client package
+
+  const individualClient = new $SERVICEClient({
+    // Input configuration with type hints
+  });
+
+  try {
+    // $OPERATIONCommandOutput generated from the Smithy model
+    const response: $OPERATIONCommandOutput = await individualClient.send(new $OPERATIONCommand({}));
+  } catch (error) {
+    // If more errors are defined in the Smithy model, then more extensive checks can be made
+    if (error instanceof $SERVICEServiceException) {
+      console.error("Oh no, a service exception was thrown!");
+    }
+    throw error;
+  }
+  ```
+- Runtime Configurations: Populated default values for a client input configuration for different platforms, currently supporting Node.js, Browser, and React-Native. All of these have a shared runtime configuration that is overwritten with more specific platform values. Not every client input configuration needs a default value, but it is best practice to provide a reasonable default. For example, the `extensions` property defaults to an empty array when no runtime extensions are specified.
+  ```text
+  Least-specific
+
+  ┌──────────────────────────────────────────────────────────┐
+  │ Shared Runtime Configuration (`runtimeConfig.shared.ts`) │
+  └──────────────────────────────────────────────────────────┘
+                 │                                   │
+  ┌──────────────────────────────┐ ┌─────────────────────────────────────-┐
+  │ Node.js (`runtimeConfig.ts`) │ │ Browser (`runtimeConfig.browser.ts`) │
+  └──────────────────────────────┘ └──────────────────────────────────────┘
+                                                     │
+                                   ┌─────────────────────────────────────────-┐
+                                   │ React-Native (`runtimeConfig.native.ts`) │
+                                   └──────────────────────────────────────────┘
+
+  Most-specific (overrides values from parent)
+  ```
+- Runtime Extensions: Interfaces to implement extensions enabling alternative default values to the runtime configuration. See [the section on customizing TypeScript Client Configuration for more details](#typescript-client-configuration).
+- Package Configuration files: Boilerplate including `package.json` and TypeScript configuration files for different platforms.
+
+Other directories could include code generated paginators (see `pagination/`), waiters, endpoint resolvers, but are usually generated only when traits are present. If code-generating custom files for the SDK client, it is recommended to use a separate directory for separation of concerns.
```

Comment:
Link to waiter trait in smithy.io (same for pagination)

Reply:
Links to the pagination and waiters docs added.